src/foam/nanos/crunch/MinMaxCapability.js

```diff
@@ -0,0 +1,80 @@
+foam.CLASS({
+  package: 'foam.nanos.crunch',
+  name: 'MinMaxCapability',
+  extends: 'foam.nanos.crunch.Capability',
+
+  javaImports: [
+    'foam.dao.ArraySink',
+    'foam.dao.DAO',
+    'java.util.List',
+    'static foam.mlang.MLang.*'
+  ],
+
+  properties: [
+    {
+      name: 'min',
+      class: 'Int',
+      value: 1,
+    },
+    {
+      name: 'max',
+      class: 'Int',
+      value: 0,
+    }
+  ],
+
+  methods: [
+    {
+      name: 'getPrereqsChainedStatus',
+      type: 'CapabilityJunctionStatus',
+      args: [
+        { name: 'x', type: 'Context' }
+      ],
+      javaCode: `
+        // Required services and DAOs
+        CrunchService crunchService = (CrunchService) x.get("crunchService");
+        DAO userCapabilityJunctionDAO = (DAO) x.get("userCapabilityJunctionDAO");
+
+        // Prepare to count statuses
+        int numberGranted = 0;
+        int numberPending = 0;
+
+        // Create ccJunctions list
+        DAO myPrerequisitesDAO = ((DAO)
+          x.get("prerequisiteCapabilityJunctionDAO"))
+            .where(
+              EQ(CapabilityCapabilityJunction.SOURCE_ID, getId()));
+        List<CapabilityCapabilityJunction> ccJunctions =
+          ((ArraySink) myPrerequisitesDAO.select(new ArraySink()))
+          .getArray();
+
+        // Count junction statuses
+        for ( CapabilityCapabilityJunction ccJunction : ccJunctions ) {
+          Capability cap = (Capability) ccJunction.findSourceId(x);
+          if ( ! cap.getEnabled() ) continue;
+
+          UserCapabilityJunction ucJunction =
+            crunchService.getJunction(x, ccJunction.getTargetId());
+          if ( ucJunction == null ) continue;
+
+          switch ( ucJunction.getStatus() ) {
+            case GRANTED:
```

Comment:
I think this switch need to check for the following as well
GRACE_PERIOD == GRANTED
APPROVED == PENDING